src/utils/I18nUtils.ts

```diff
@@ -0,0 +1,27 @@
+import { Alert, NativeModules } from 'react-native';
+import RNRestart from 'react-native-restart';
+import rtlDetect from 'rtl-detect';
+
+export const languageRestart = async (prevLang, lang) => {
+  if (prevLang != lang) {
+    // if selected lang is RTL, switch the layout
+    if (rtlDetect.isRtlLang(lang)) {
```

Comment:
may be we should check if layout needs to change based on previous language and only restart if it is required... it seems right now it is asking for restart even layout do not require change

Reply:
updated. https://github.com/ecency/ecency-mobile/pull/2116/commits/79539637755f2133239ab47f43ff49fcb3b7e6ea